src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  let result = sourceString.split(';').map((arr) => arr.split(':'));
```

Comment:
u do 3 iterations(map -> filter -> map) over the entire line or array. 
u can do all this in the middle of your reduce method
```suggestion
  const splitedString = sourceString.split(';');
```